icubam/analytics/preprocessing.py

```diff
@@ -0,0 +1,350 @@
+import itertools
+import logging
+
+import numpy as np
+import pandas as pd
+
+CUM_COLUMNS = [
+  "n_covid_deaths",
+  "n_covid_healed",
+  "n_covid_transfered",
+  "n_covid_refused",
+]
+NCUM_COLUMNS = [
+  "n_covid_free",
+  "n_ncovid_free",
+  "n_covid_occ",
+  "n_ncovid_occ",
+]
+BEDCOUNT_COLUMNS = CUM_COLUMNS + NCUM_COLUMNS
+ALL_COLUMNS = ([
+  "icu_name", "create_date", "datetime", "icu_dept", "icu_region_id",
+  "icu_region_name"
+] + CUM_COLUMNS + NCUM_COLUMNS)
+
+SPREAD_CUM_JUMPS_MAX_JUMP = {
+  "n_covid_deaths": 10,
+  "n_covid_transfered": 10,
+  "n_covid_refused": 10,
+  "n_covid_healed": 10,
+}
+
+
+def format_data(d):
+  d["datetime"] = pd.to_datetime(d["date"])
+  d["date"] = d["datetime"].dt.date
+  d = d[ALL_COLUMNS]
+  return d
+
+
+def preprocess_data(
```

Comment:
So this is preprocesing mostly unchanged but with additional comments from the predicu file, right?

Reply:
Remove as much of FR-specific processing, more comments, and only ICUBAM data.